resource-managers/mesos/src/test/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackendSuite.scala

```diff
@@ -544,6 +543,21 @@ class MesosCoarseGrainedSchedulerBackendSuite extends SparkFunSuite
     assert(launchedTasks.head.getName == "test-mesos-dynamic-alloc 0")
   }
 
+  test("mesos sets different task ids across executions") {
```

Comment:
Would this not have been the case before? Only that the first ID would be `0` and the second `1`?

Reply:
A driver will assign task ids starting from 0 after `SparkContext` restart. This test stops the Context and restarts it thus verifying that the ids are different for the different driver instances which was the original issue. Adding explicit assertion like `assert(taskId != "0")` doesn't seem reasonable to me because it involves some kind of historical knowledge which wouldn't be obvious for the future contributors.